src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,34 @@
 package core.basesyntax;
 
 public class MainApp {
+    public MainApp(String[] machine) {
 
+        Machine truck = new Truck() {
+            @Override
+
+            public void doWork() {
+                super.doWork();
+            }
+        };
```

Comment:
Using abstract references instead of specific ones where possible allows for more flexible code. Consider using `Machine truck = new Truck();` without the anonymous class.